src/axom/core/tests/core_openmp_map.hpp

```diff
@@ -0,0 +1,275 @@
+// Copyright (c) 2017-2021, Lawrence Livermore National Security, LLC and
+// other Axom Project Developers. See the top-level LICENSE file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+#include "axom/core/execution/execution_space.hpp"
+#include "axom/core/execution/for_all.hpp"
+#include "axom/core/Map.hpp"
+#include "gtest/gtest.h" /* for TEST and EXPECT_* macros */
+#include <string>
+
+namespace axom
+{
+namespace internal
+{
+template <typename Key, typename T, typename Hash, typename Policy>
+experimental::Map<Key, T, Hash, Policy> init(int N, int len)
+{
+  experimental::Map<Key, T, Hash, Policy> test(N, len);
+  EXPECT_EQ(N * len, test.max_size());
+  EXPECT_EQ(0, test.size());
+  EXPECT_EQ(true, test.empty());
+  EXPECT_EQ(len, test.bucket_size());
+  EXPECT_EQ(N, test.bucket_count());
+  return test;
+}
+
+template <typename Key, typename T, typename Hash, typename Policy>
+void test_storage(experimental::Map<Key, T, Hash, Policy> &test)
+{
+  experimental::Map<Key, T, Hash, Policy> *test2 = &test;
+  axom::for_all<axom::OMP_EXEC>(
+    0,
+    test.max_size(),
+    AXOM_LAMBDA(IndexType idx) {
+      Key key = idx;
+      T value = key * 27;
+      auto ret_test = test2->insert(key, value);
+      EXPECT_EQ(true, ret_test.second);
+    });
+  EXPECT_EQ(false, test.empty());
+  axom::for_all<axom::OMP_EXEC>(
```

Comment:
Please refactor the tests in this file w.r.t. the policy: 
They should *either* use the templated `Policy` *or* be explicitly written for `axom::OMP_EXEC`.
As is, it seems to be in a middle ground -- having a `Policy` but not using it.